Format subject dates according to instance setting

NobodysNightmare opened the pull request:
We default to the ISO8601 format, if the instance
set no preference, but otherwise we use the instance's preference.

# Ticket
https://community.openproject.org/wp/63481

NobodysNightmare committed:
Format dates according to instance setting

We default to the ISO8601 format, if the instance
set no preference, but otherwise we use the instance's
preference.

Kharonus commented on app/models/work_package_types/patterns/token_property_mapper.rb:
```diff
@@ -33,7 +33,7 @@ module Patterns
       STRING_OR_NIL = ->(v) { v&.to_s }
       ARRAY = ->(v) { v.compact.presence&.join(", ") }
-      DATE = ->(v) { v&.strftime("%Y-%m-%d") }
+      DATE = ->(v) { v&.strftime(Setting.date_format || "%Y-%m-%d") }
```
🟢 came in very handy to have that formatter in here now. :D

Kharonus commented on frontend/src/stimulus/controllers/dynamic/documents/live-events.controller.ts:
⚠️ what?? where does those changes come from? They seem utterly unrelated to your scope of the bugfix. Can you double check, if something went wrong with merging or rebasing?

Kharonus replied:
looking at the commit history, those are Bruno's commits. and probably when retargeting the PR, this is the leftover.

NobodysNightmare replied:
yeah, it's still surprising me a bit. But I rebased them out.

Kharonus reviewed (approved):
Much better!

NobodysNightmare commented:
Those specs are failing extremely consistently on this branch:

```
rspec ./spec/features/work_packages/details/relations/hierarchy_spec.rb:64 # within a primerized split screen behaves like work package relations tab as admin allows to manage hierarchy
rspec ./spec/features/work_packages/details/relations/hierarchy_spec.rb:64 # within a full screen behaves like work package relations tab as admin allows to manage hierarchy
rspec ./spec/features/work_packages/details/relations/hierarchy_spec.rb:64 # within a split screen behaves like work package relations tab as admin allows to manage hierarchy
```

I can't reproduce their failure locally, but I also don't see how they are related to the changes here at all :disappointed:

NobodysNightmare committed:
Fix flickering spec

Selection by work package ID was unreliable, because the text
that's selected by is from anywhere in the selected element.

The element shows different information:
* WP subject
* WP ID
* Project name
* status name

The intention of the original code was to uniquely select work packages
by their ID, however it was also perfectly possible to have a project
named "Project 123" as the parent of all work packages. So if the WP
id happens to be 123 as well, the selection would be effectively random
across the work packages.

(Credit for discovering the root cause of this flicker goes to Christophe Bliard
if anyone is asking ;-) )

Fixes flickers of spec/features/work_packages/details/relations/hierarchy_spec.rb,

Reproduction instructions as provided by Christophe:

    rm -f db/structure.sql; \
    bin/rails db:drop db:create db:migrate RAILS_ENV=test > /dev/null ; \
    CI=true TZ=UTC OPENPROJECT_LOG__LEVEL=info CAPYBARA_PUMA_THREADS=0:4 DISABLE_SPRING=1 \
    bin/rspec --seed 34872 -fd \
    'spec/features/work_packages/details/relations/hierarchy_spec.rb[1:1:1:1]' \
    spec/features/work_packages/work_package_index_spec.rb:46

cbliard reviewed (approved):
Fix of flickering tests is 👌
